src/transformers/models/perceiver/modeling_perceiver.py

```diff
@@ -0,0 +1,2769 @@
+            Classification (or regression if config.num_labels==1) scores (before SoftMax).
+        last_hidden_state (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, sequence_length, hidden_size)`):
+            Sequence of hidden-states at the output of the last layer of the model.
+        hidden_states (:obj:`tuple(torch.FloatTensor)`, `optional`, returned when ``output_hidden_states=True`` is passed or when ``config.output_hidden_states=True``):
+            Tuple of :obj:`torch.FloatTensor` (one for the output of the embeddings + one for the output of each layer)
+            of shape :obj:`(batch_size, sequence_length, hidden_size)`. Hidden-states of the model at the output of
+            each layer plus the initial embedding outputs.
+        attentions (:obj:`tuple(torch.FloatTensor)`, `optional`, returned when ``output_attentions=True`` is passed or when ``config.output_attentions=True``):
+            Tuple of :obj:`torch.FloatTensor` (one for each layer) of shape :obj:`(batch_size, num_heads,
+            sequence_length, sequence_length)`. Attentions weights after the attention softmax, used to compute the
+            weighted average in the self-attention heads.
+        cross_attentions (:obj:`tuple(torch.FloatTensor)`, `optional`, returned when ``output_attentions=True`` is passed or when ``config.output_attentions=True``):
+            Tuple of :obj:`torch.FloatTensor` (one for each layer) of shape :obj:`(batch_size, num_heads,
+            sequence_length, sequence_length)`. Attentions weights of the decoder's cross-attention layer, after the
+            attention softmax, used to compute the weighted average in the cross-attention heads.
+    """
+
+    logits: torch.FloatTensor = None
+    last_hidden_state: torch.FloatTensor = None
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    attentions: Optional[Tuple[torch.FloatTensor]] = None
+    cross_attentions: Optional[Tuple[torch.FloatTensor]] = None
+
+
+@dataclass
+class PerceiverDecoderOutput(ModelOutput):
+    """
+    Base class for Perceiver decoder outputs, with potential cross-attentions.
+
+    Args:
+        logits (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, num_labels)`):
+            Output of the basic decoder.
+        cross_attentions (:obj:`tuple(torch.FloatTensor)`, `optional`, returned when ``output_attentions=True`` is passed or when ``config.output_attentions=True``):
+            Tuple of :obj:`torch.FloatTensor` (one for each layer) of shape :obj:`(batch_size, num_heads,
+            sequence_length, sequence_length)`. Attentions weights of the decoder's cross-attention layer, after the
+            attention softmax, used to compute the weighted average in the cross-attention heads.
+    """
+
+    logits: torch.FloatTensor = None
+    cross_attentions: Optional[Tuple[torch.FloatTensor]] = None
+
+
+@dataclass
+class PerceiverMaskedLMOutput(MaskedLMOutput):
```

Comment:
not a fan of this abstraction - why not copy paste it here? I haven't seen this being done before - what do you think @sgugger ?

Reply:
It only takes the copy of the four arguments, so it's easy to add here without depending on `MaskedLMOutput`, but I think we discussed with @LysandreJik that it would be better to have the model outputs use inheritances so we can make some checks on the classes. I don't remember the specific however.

Reply:
@patrickvonplaten I've actually done this for quite some models now, e.g. [TAPAS](https://github.com/huggingface/transformers/blob/e4c67d60ec07ac6402951848c0e65639e81b65fe/src/transformers/models/tapas/modeling_tapas.py#L106), [CANINE](https://github.com/huggingface/transformers/blob/e4c67d60ec07ac6402951848c0e65639e81b65fe/src/transformers/models/canine/modeling_canine.py#L66)

Reply:
Oh I see, you mean it's better to copy all attributes, I can do that.

Reply:
I would prefer this, but let's see what @LysandreJik says

Reply:
I've fixed it, copied all arguments.